app/actions/handlers.py

```diff
@@ -183,26 +191,19 @@ async def action_pull_events(integration: Integration, action_config: PullEvents
 
         await state_manager.set_geostores_id_ttl(aoi_data.id, 86400*7)
 
-    # This semaphore is meant to limit the concurrent requests to GFW's dataset API query endpoints.
-    # When configuring a cloud run service, include this in a calculation so that 
-    # GFW_DATASET_QUERY_CONCURRENCY * maximum-number-of-instances * maximum-concurrent-requests-per-instance <= N
-    # where N is the maximum concurrent requests allowed by GFW's API. 
-    # (ex. in practice, N is around 50)
-    sema = asyncio.Semaphore(app.settings.GFW_DATASET_QUERY_CONCURRENCY)
-
-    # Create a list of tasks.
-    tasklist = [get_nasa_viirs_fire_alerts(integration, action_config, auth_config, aoi_data, sema),
-                 get_integrated_alerts(integration, action_config, auth_config, aoi_data, sema)
-                 ]
-    
-    # Wait until they're all finished.
-    tasks_results = await asyncio.gather(*tasklist)
+    # Trigger "get_dataset_and_geostores" sub-action
+    config = GetDatasetAndGeostoresConfig(
+        integration_id=str(integration.id),
+        pull_events_config=action_config,
+        auth_config=auth_config,
```

Comment:
This should not be necessary (and I think it will not work as-is).
I recommend removing the `auth_config` property from this config model and let the actions use `get_auth_config`  instead.